base/strings/search.jl

```diff
@@ -212,44 +206,50 @@ julia> rsearch("aaabbb","b")
 6:6
 ```
 """
-function rsearch(s::AbstractString, c::Chars, i::Integer)
-    e = endof(s)
-    j = search(RevString(s), c, e-i+1)
-    j == 0 && return 0
-    e-j+1
+function rsearch(s::AbstractString, c::Chars, i::Integer=start(s))
+    @boundscheck checkbounds(s, i)
+    isempty(c) && return i
+    j = i
+    while j > 0
+        d = s[j]
+        d in c && return j
```

Comment:
Not sure why you need the variable `d` here, as opposed to `s[j] in c && return j`